### PR TITLE
Adding deployment files for running on capz with calico CNI

### LINKED ISSUE #2: Create HostProcessContainer files for KPNG
Create the Dockerfile.windows, building scripts and specs to allow running the KPNG module as HPC on a cluster.

/assign @marosset

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

Fixes #2

### CODE REVIEW

**Comment by marosset** [comment]
/hold until we can get an image published to the staging repo 
https://kubernetes.slack.com/archives/CCK68P2Q2/p1674259965028119

File: deploy/start-kpng.ps1 [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+Write-Host "Importing hns.psm1"
+Import-Module C:\hpc\hns.psm1
+
+$NetworkName = $env:NETWORK_NAME
+Write-Host "Waiting for network '$NetworkName' to be available..."
+while (-Not (Get-HnsNetwork | ? Name -EQ $NetworkName)) {
+    Write-Debug "waiting for HNS network..."
+    Start-Sleep 5
+}
+Write-Host "Found HNS network '$NetworkName'"
```
Comment (jsturtevant) [inline deploy/start-kpng.ps1:19]: We should attempt to remove the requirement for a powershell script and move this check into kubelet. We have a check there when it is overlay, not sure if it is required for non-overlay networks.

File: deploy/start-kpng.ps1 [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+Write-Host "Network type: $network.Type"
+
+if ($network.Type -EQ "overlay") {
+    WRite-Host "Overlay / VXLAN network detected. Waiting for host endpoint to be created..."
+    while( -not (Get-HnsEndpoint | ? Name -EQ "${NetworkName}_ep")) {
+        Start-Sleep 1
+    }
+    $sourceVip = (Get-HnsEndpoint | ? Name -EQ "${NetworkName}_ep").IpAddress
+    Write-Host "Host endpoint found. Source CIP: $sourceVip"
+    $argList += "--source-vip=$sourceVip"
```
Comment (jsturtevant) [inline deploy/start-kpng.ps1:40]: same here using a well known name as proposed in https://docs.google.com/document/d/1A6Gkyx5EvL86Z4L2P4sxnk9HQjdRgnLXvBg6tIMDB3s/edit#heading=h.bkxs3rpgbz7g

**Review by jsturtevant — COMMENTED** [review]
/lgtm

I would like to eventually remove the need for powershell scripts so we don't need to sign those and package them along with the rest of the image.  There are only two cases currently that require it, but we should be able to remove them.

**Comment by knabben** [comment]
> /lgtm
> 
> I would like to eventually remove the need for powershell scripts so we don't need to sign those and package them along with the rest of the image.  There are only two cases currently that require it, but we should be able to remove them.

Should we have other ticket for it? Or it must be  fixed on k/k or sig windows tools repo first?

File: deploy/start-kpng.ps1 (hunk 1 above)
Comment (marosset) [reply deploy/start-kpng.ps1:19]: Can we do this in another PR?
I'll create an issue to track the work.

Comment (jsturtevant) [reply deploy/start-kpng.ps1:19]: agree not with in scope of this work, just wanted to bring attention to it.  Thanks!

Comment (marosset) [reply deploy/start-kpng.ps1:19]: https://github.com/kubernetes-sigs/windows-service-proxy/issues/27 filed!

**Comment by marosset** [comment]
/hold cancel

We how have an image building and published to `gcr.io/k8s-staging-win-svc-proxy/kpng:latest`

**Review by knabben — COMMENTED** [review]
/lgtm

**Pull request merged**